src/v/bytes/iobuf_istreambuf.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "bytes/iobuf.h"
+
+#include <streambuf>
+
+/// A simple istream buffer reader.
+///
+/// iobuf underlying;
+/// std::string out;
+/// iobuf_istreambuf ibuf(underlying);
+/// std::istream is(&ibuf);
+///
+/// is >> out;
+///
+class iobuf_istreambuf final : public std::streambuf {
+public:
+    explicit iobuf_istreambuf(iobuf& buf) noexcept
+      : _begin(buf.begin())
+      , _curr(buf.begin())
+      , _end(buf.end()) {}
+    iobuf_istreambuf(iobuf_istreambuf&&) noexcept = default;
+    iobuf_istreambuf& operator=(iobuf_istreambuf&&) noexcept = default;
+    iobuf_istreambuf(const iobuf_istreambuf&) = delete;
+    iobuf_istreambuf& operator=(const iobuf_istreambuf&) = delete;
+    ~iobuf_istreambuf() noexcept override = default;
+
+    int_type underflow() final {
```

Comment:
if class is final, no need to mark underflow as final.

Reply:
clang-tidy complains when there is no keyword so I changed it to 'override'